src/main/java/net/imglib2/img/basictypeaccess/Accesses.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * #%L
+ * ImgLib2: a general-purpose, multidimensional image processing library.
+ * %%
+ * Copyright (C) 2009 - 2018 Tobias Pietzsch, Stephan Preibisch, Stephan Saalfeld,
+ * John Bogovic, Albert Cardona, Barry DeZonia, Christian Dietz, Jan Funke,
+ * Aivar Grislis, Jonathan Hale, Grant Harris, Stefan Helfrich, Mark Hiner,
+ * Martin Horn, Steffen Jaensch, Lee Kamentsky, Larry Lindsey, Melissa Linkert,
+ * Mark Longair, Brian Northan, Nick Perry, Curtis Rueden, Johannes Schindelin,
+ * Jean-Yves Tinevez and Michael Zinsmaier.
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package net.imglib2.img.basictypeaccess;
+
+/**
+ * Utility and helper methods for accesses ({@link ByteAccess} etc).
+ *
+ * @author Philipp Hanslovsky
+ */
+public interface Accesses
+{
+
+	public static void copy( final ByteAccess source, final ByteAccess target, final int size )
+	{
+		copy( source, target, 0, size );
+	}
+
+	public static void copy( final ByteAccess source, final ByteAccess target, final int start, final int stop )
```

Comment:
This signature assumes the offset into `source` and `target` will be the same, which may not necessarily hold. I suggest modeling the API after `System.arraycopy`, which supports separate offsets for the source and dest arrays. So in this case: `copy(ByteAccess src, int srcPos, ByteAccess dest, int destPos, int length)`.

Reply:
Good point. I wonder how complext this should be (you could even think of a different stride). Maybe it's best to just have `copy( source, target, size )`, then.

Reply:
My preference would be to match the `System.arraycopy` method. It will be familiar to Java programmers, and is more powerful.

Reply:
would be my preference too

Reply:
Will mimic the `System.arraycopy` method then.